ansible/roles/k8s_master/tasks/init_master_leader.yml

```diff
@@ -48,3 +48,18 @@
     dest: /home/ubuntu/.kube/config
     remote_src: yes
     owner: ubuntu
+ 
+- name: Helper for HAProxy to have copy of admin.conf
+  become: yes
+  fetch:
+    src: /etc/kubernetes/admin.conf
+    dest: /tmp/helper/admin.conf
```

Comment:
>/tmp/helper/ [](start = 10, length = 12)

Why not set `dest` to `/var/www/html` directly?

Reply:
This saves the file to host server so that the file can be transferred from one VM to another VM. init_master_leader.yml is running on Master1 VM, but I need the file on the HAProxy VM

Reply:
Then you may remove this `fetch` completely, and use below `copy` with `src` as `/etc/kubernetes/admin.conf` ?

---
In reply to: [508867244](https://github.com/Azure/sonic-mgmt/pull/2306#discussion_r508867244) [](ancestors = 508867244)

Reply:
Then I get this error: 
`"Could not find or access '/etc/kubernetes/admin.conf' on the Ansible Controller.` When I wrote the original code I also thought there would be a more straightforward way to copy the file from one VM to another, but using the /tmp directory was the only way I tried that worked for me

Reply:
I understand now. Please give correct comments, especially on the src/dest machines.

---
In reply to: [508884204](https://github.com/Azure/sonic-mgmt/pull/2306#discussion_r508884204) [](ancestors = 508884204,508867244)

Reply:
To clarify, I get that error because the `/etc/kubernetes/admin.conf` is on the server, not on the VM. 
For fetch module:

- `src` is the master VM, on which `/etc/kubernetes/admin.conf` is present
- `dest` is the server

For copy module: 
- `src` is the server
- `dest` is the HAProxy VM